lib/this_data/track_request.rb

```diff
@@ -20,19 +20,32 @@ class ThisDataTrackError < StandardError; end
     #       `current_user`.
     #     The object must respond to at least
     #       `ThisData.configuration.user_id_method`, which defaults to `id`.
+    #   authenticated: (Boolean, Optional, default nil). If the user being
+    #     tracked has not yet authenticated successfully, set this to false.
+    #     This is only needed if passing a user manually and that user hasn't
+    #     logged in yet (e.g. a log-in-denied event), or if user_method can
+    #     return an unauthenticated user.
```

Comment:
An alternative way to write this could be:
"Used to indicate whether a user is authenticated or not. By default we assume that if there is a user specified, they are authenticated, but in some cases (like a log-in-denied event) you might want to track the user but tell us that they are not authenticated. In these situations you should set `authenticated` to false."

Just thinking that the current explanation doesn't really convey the intent of the attribute.